sql/core/src/test/scala/org/apache/spark/sql/DataFrameWriterV2Suite.scala

```diff
@@ -839,4 +839,26 @@ class DataFrameWriterV2Suite extends QueryTest with SharedSparkSession with Befo
       condition = "CALL_ON_STREAMING_DATASET_UNSUPPORTED",
       parameters = Map("methodName" -> "`writeTo`"))
   }
+
+  test("SPARK-51281: create/replace file source tables") {
+    Seq(true, false).foreach { ignorePath =>
+      withSQLConf(SQLConf.LEGACY_DF_WRITER_V2_IGNORE_PATH_OPTION.key -> ignorePath.toString) {
+        withTable("t1", "t2") {
+          spark.range(10).writeTo("t1").using("json").create()
+          checkAnswer(spark.table("t1"), spark.range(10).toDF())
```

Comment:
`spark.range(10).toDF()` seems to be repeated three times. Shall we define one to reuse during this unit test?